### ISSUE TITLE
Development plan for ESPnet2 speaker verification (SID)

### ISSUE THREAD

**Issue by underdogliu** [espnet/espnet#3421]
This is an issue for developing a decent framework for speaker verification under ESPnet2. 

We sometimes call it 'speaker id' but just in case, on the evaluation side identification and verification are not the same thing XD.

The very prior aim is to generate speaker embeddings which are compatible with other tasks such as TTS and ST, especially for TTS where speaker embeddings are playing a crucial part.

As a starting point, we will start from an x-vector system with PLDA scoring backend. But definitely we shall acquire more advanced networks and scoring backends and strategies.

We list the minimal feature under development below and welcome feature requests and new ideas. Please contact @underdogliu if having any question (or complaint).

### Progress & Plan
- [ ] Minimal X-vector based system
- [ ] A new task for speaker verification
- [ ] Template and recipe based on VoxCeleb

### Network Architectures:
- [ ] [X-Vector](http://www.danielpovey.com/files/2017_interspeech_embeddings.pdf)
- [ ] [Extended X-Vector](https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=8683760)
- [ ] [ECAPA-TDNN](https://arxiv.org/abs/2005.07143)

### Pooling
- [ ] [Statistics Pooling](http://www.danielpovey.com/files/2017_interspeech_embeddings.pdf)
- [ ] [Attentive Statistics Pooling](https://arxiv.org/pdf/1803.10963v1.pdf)

### Loss
- [ ] Softmax
- [ ] [AAM-Softmax Loss](https://arxiv.org/pdf/1801.07698v1.pdf)
- [ ] [Generalised E2E](https://arxiv.org/pdf/1710.10467.pdf) [source code](https://github.com/cvqluu/GE2E-Loss)

### Documentation
 - [ ] add documentation in https://github.com/espnet/espnet/tree/master/doc
 - [ ] prepare the webpage to put some demonstration
 - [ ] create notebook for the demo

### Recipe
- [ ] VoxCeleb1
- [ ] VoxCeleb2 (evaluation: VoxCeleb1-E/H)
- [ ] Speakers in the Wild (SITW)

### Misc.
- [ ] ASR as evaluation metric
- [ ] More loss functions
- [ ] Integration test

**Comment by underdogliu** [comment]
@ftshijt @sw005320 labels are needed. Could you please do that? (I'm not a contributor yet. Will delete this message after the adding)

**Comment by sw005320** [comment]
Cool!
I just created a SID label as well!

**Comment by underdogliu** [comment]
@sw005320 I am really sorry for delaying this project for so long time due to last year's busy schedule. But now to kick off I am back and will put decent effort on this. 

To kick off, I do have one question. How things work when we would like to adopt solutions from other toolkits? I did implement a PLDA backend for espnet2 from my own side, but the performance was not very promising and recently there are speechbrain who has a backend and scoring scheme that can reach good performance [link](https://github.com/speechbrain/speechbrain/blob/develop/recipes/VoxCeleb/SpeakerRec/speaker_verification_plda.py). I would like to kinda adopt their code and see how it will work. Is it legitimate here as long as I keep the copyright header (their code is Apache 2.0, as shown)?

**Comment by sw005320** [comment]
Sounds good to me.
Thanks for sharing your status.

**Comment by underdogliu** [comment]
Hi @sw005320 and hope you are doing well starting 2023. So I had a quick recap of this PR after a REALLY long-delayed period and again sorry for not finding decent amount of time working on this base repo - I feel shamed of myself on this and I hope I can be forgiven.......

This PR was drafted at 2021 and since then a lot of more advanced techniques in SID and speaker verification have been discovered, including ECAPA-TDNN, RawNet, and self-supervised learning thingy. I had a quick talk with an upcoming new member in the WavLab group @Jungjee and we agree that it would be a good idea to close this plan and move on to RawNet-based ones (indicated by PRs such as #4884) which are originated by him (@Jungjee, correct me if there is any info wrongly-stated).

I will keep looking at espnet2 and contribute if necessary since I am moving on from common speaker verification (which is my PhD thesis about) to other speech processing task and the application of speaker/token-wise information into it.
